patches/tModLoader/Terraria/WorldGen.cs.patch

```diff
@@ -975,7 +975,14 @@
  			for (int num7 = num - 1; num7 < num + 3; num7++) {
  				for (int num8 = num2 - 1; num8 < num2 + 3; num8++) {
  					TileFrame(num7, num8);
-@@ -31181,9 +_,6 @@
+@@ -31176,14 +_,13 @@
+ 					tileSafely.frameX = (short)(k * 18);
+ 					tileSafely.frameY = (short)(style * num + l * 18);
+ 					tileSafely.type = type;
++					if (type == 172)
++						tileSafely.countsAsWaterSource = true;
```

Comment:
I think this edit wasn't reverted from the old approach

Reply:
fixed